shims/src/peripherals/output.rs

```diff
@@ -36,26 +37,30 @@ impl<'a, R: ?Sized> DerefMut for OwnedOrRef<'a, R> {
     }
 }
 
-pub struct OutputShim<'a> {
+pub struct OutputShim<'a, 'b> {
     // pub struct OutputShim {
     // sink: &'a mut dyn Write,
     // sink: Box<dyn Write>,
     sink: OwnedOrRef<'a, dyn Write + 'a>,
     // _marker: PhantomData<&'a ()>,
     // sink: &'a mut dyn AsRef<dyn Write>,
+    flag: Option<&'b AtomicBool>,
```

Comment:
Why does this have a separate lifetime?

Reply:
Wasn't sure what lifetime the sink needed so I left them separate. Otherwise the sink will have to be static on the board and idk what lifetime it should be. Not hard to change though